man/man/man3/ConfigMisc.3
.TH "ConfigMisc" 3 "Thu Aug 23 2012" "Version 1.0" "FS-UAE Gui for Linux OS" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigMisc \- 
.PP
This class is responsible for the Miscellaneous Configuration management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <configmisc\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigMisc\fP ()"
.br
.ti -1c
.RI "string \fBgetInputGrabConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetInputGrabString\fP ()"
.br
.ti -1c
.RI "string \fBgetAutomaticInputGrabConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetAutomaticInputGrabString\fP ()"
.br
.ti -1c
.RI "string \fBgetBsdSocketLibraryConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetBsdSocketLibraryString\fP ()"
.br
.ti -1c
.RI "string \fBgetAudioBufferConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetAudioBufferString\fP ()"
.br
.ti -1c
.RI "string \fBgetMouseSpeedConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetMouseSpeedString\fP ()"
.br
.ti -1c
.RI "string \fBgetTitleConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetTitleString\fP ()"
.br
.ti -1c
.RI "string \fBgetSubTitleConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetSubTitleString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayServerConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayServerString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayPortConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayPortString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayTagConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayTagString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayPasswordConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetNetPlayPasswordString\fP ()"
.br
.ti -1c
.RI "string \fBgetBaseDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetBaseDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetKickstartsDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetKickstartsDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetSaveStatesDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetSaveStatesDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloppyOverlaysDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloppyOverlaysDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetFlashMemoryDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFlashMemoryDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetControllersDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetControllersDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetLogsDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetLogsDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetHardDrivesDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetHardDrivesDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetCDRomsDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetCDRomsDirString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloppiesDirConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloppiesDirString\fP ()"
.br
.ti -1c
.RI "int \fBsetParameter\fP (string parameter, string value)"
.br
.ti -1c
.RI "void \fBsetToDefaultConfiguration\fP ()"
.br
.ti -1c
.RI "bool \fBhasParameter\fP (string parameterName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "string \fBinput_grab\fP"
.br
.ti -1c
.RI "string \fBautomatic_input_grab\fP"
.br
.ti -1c
.RI "string \fBbsdsocket_library\fP"
.br
.ti -1c
.RI "string \fBaudio_buffer_target_bytes\fP"
.br
.ti -1c
.RI "string \fBmouse_speed\fP"
.br
.ti -1c
.RI "string \fBtitle\fP"
.br
.ti -1c
.RI "string \fBsub_title\fP"
.br
.ti -1c
.RI "string \fBnetplay_server\fP"
.br
.ti -1c
.RI "string \fBnetplay_port\fP"
.br
.ti -1c
.RI "string \fBnetplay_tag\fP"
.br
.ti -1c
.RI "string \fBnetplay_password\fP"
.br
.ti -1c
.RI "string \fBbase_dir\fP"
.br
.ti -1c
.RI "string \fBkickstarts_dir\fP"
.br
.ti -1c
.RI "string \fBsave_states_dir\fP"
.br
.ti -1c
.RI "string \fBfloppy_overlays_dir\fP"
.br
.ti -1c
.RI "string \fBflash_memory_dir\fP"
.br
.ti -1c
.RI "string \fBcontrollers_dir\fP"
.br
.ti -1c
.RI "string \fBlogs_dir\fP"
.br
.ti -1c
.RI "string \fBhard_drives_dir\fP"
.br
.ti -1c
.RI "string \fBcdroms_dir\fP"
.br
.ti -1c
.RI "string \fBfloppies_dir\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class is responsible for the Miscellaneous Configuration management\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "\fBConfigMisc::ConfigMisc\fP ()"
.SH "Member Function Documentation"
.PP 
.SS "string \fBConfigMisc::getAudioBufferConfigString\fP ()"
.SS "string \fBConfigMisc::getAudioBufferString\fP ()"
.SS "string \fBConfigMisc::getAutomaticInputGrabConfigString\fP ()"
.SS "string \fBConfigMisc::getAutomaticInputGrabString\fP ()"
.SS "string \fBConfigMisc::getBaseDirConfigString\fP ()"
.SS "string \fBConfigMisc::getBaseDirString\fP ()"
.SS "string \fBConfigMisc::getBsdSocketLibraryConfigString\fP ()"
.SS "string \fBConfigMisc::getBsdSocketLibraryString\fP ()"
.SS "string \fBConfigMisc::getCDRomsDirConfigString\fP ()"
.SS "string \fBConfigMisc::getCDRomsDirString\fP ()"
.SS "string \fBConfigMisc::getControllersDirConfigString\fP ()"
.SS "string \fBConfigMisc::getControllersDirString\fP ()"
.SS "string \fBConfigMisc::getFlashMemoryDirConfigString\fP ()"
.SS "string \fBConfigMisc::getFlashMemoryDirString\fP ()"
.SS "string \fBConfigMisc::getFloppiesDirConfigString\fP ()"
.SS "string \fBConfigMisc::getFloppiesDirString\fP ()"
.SS "string \fBConfigMisc::getFloppyOverlaysDirConfigString\fP ()"
.SS "string \fBConfigMisc::getFloppyOverlaysDirString\fP ()"
.SS "string \fBConfigMisc::getHardDrivesDirConfigString\fP ()"
.SS "string \fBConfigMisc::getHardDrivesDirString\fP ()"
.SS "string \fBConfigMisc::getInputGrabConfigString\fP ()"
.SS "string \fBConfigMisc::getInputGrabString\fP ()"
.SS "string \fBConfigMisc::getKickstartsDirConfigString\fP ()"
.SS "string \fBConfigMisc::getKickstartsDirString\fP ()"
.SS "string \fBConfigMisc::getLogsDirConfigString\fP ()"
.SS "string \fBConfigMisc::getLogsDirString\fP ()"
.SS "string \fBConfigMisc::getMouseSpeedConfigString\fP ()"
.SS "string \fBConfigMisc::getMouseSpeedString\fP ()"
.SS "string \fBConfigMisc::getNetPlayPasswordConfigString\fP ()"
.SS "string \fBConfigMisc::getNetPlayPasswordString\fP ()"
.SS "string \fBConfigMisc::getNetPlayPortConfigString\fP ()"
.SS "string \fBConfigMisc::getNetPlayPortString\fP ()"
.SS "string \fBConfigMisc::getNetPlayServerConfigString\fP ()"
.SS "string \fBConfigMisc::getNetPlayServerString\fP ()"
.SS "string \fBConfigMisc::getNetPlayTagConfigString\fP ()"
.SS "string \fBConfigMisc::getNetPlayTagString\fP ()"
.SS "string \fBConfigMisc::getSaveStatesDirConfigString\fP ()"
.SS "string \fBConfigMisc::getSaveStatesDirString\fP ()"
.SS "string \fBConfigMisc::getSubTitleConfigString\fP ()"
.SS "string \fBConfigMisc::getSubTitleString\fP ()"
.SS "string \fBConfigMisc::getTitleConfigString\fP ()"
.SS "string \fBConfigMisc::getTitleString\fP ()"
.SS "bool \fBConfigMisc::hasParameter\fP (stringparameterName)"
.SS "int \fBConfigMisc::setParameter\fP (stringparameter, stringvalue)"
.SS "void \fBConfigMisc::setToDefaultConfiguration\fP ()"
.SH "Member Data Documentation"
.PP 
.SS "string \fBConfigMisc::audio_buffer_target_bytes\fP\fC [private]\fP"
.SS "string \fBConfigMisc::automatic_input_grab\fP\fC [private]\fP"
.SS "string \fBConfigMisc::base_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::bsdsocket_library\fP\fC [private]\fP"
.SS "string \fBConfigMisc::cdroms_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::controllers_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::flash_memory_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::floppies_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::floppy_overlays_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::hard_drives_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::input_grab\fP\fC [private]\fP"
.SS "string \fBConfigMisc::kickstarts_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::logs_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::mouse_speed\fP\fC [private]\fP"
.SS "string \fBConfigMisc::netplay_password\fP\fC [private]\fP"
.SS "string \fBConfigMisc::netplay_port\fP\fC [private]\fP"
.SS "string \fBConfigMisc::netplay_server\fP\fC [private]\fP"
.SS "string \fBConfigMisc::netplay_tag\fP\fC [private]\fP"
.SS "string \fBConfigMisc::save_states_dir\fP\fC [private]\fP"
.SS "string \fBConfigMisc::sub_title\fP\fC [private]\fP"
.SS "string \fBConfigMisc::title\fP\fC [private]\fP"

.SH "Author"
.PP 
Generated automatically by Doxygen for FS-UAE Gui for Linux OS from the source code\&.


configmisc.cpp
/*! \file configmisc.cpp
 configmisc.cpp

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 Written by Riva Alessandro Mario, alessandromario.riva@gmail.com
 and Sanvito Davide, dsanvito90@gmail.com. */
#include "configmisc.h"
#include "amiga.h"
#include <QString>
#include <sstream>

const string DEFAULTINPUTGRAB = "1";
const string DEFAULTAUTOMATICINPUTGRAB = "1";
const string DEFAULTBSDSOCKET = "0";
const string DEFAULTAUDIOBUFFER = "8192";
const string DEFAULTMOUSESPEED = "100";
const string DEFAULTTITLE = "";
const string DEFAULTSUBTITLE = "";

const string DEFAULTNETPLAYSERVER = "";
const string DEFAULTNETPLAYPORT = "25100";
const string DEFAULTNETPLAYTAG = "";
const string DEFAULTNETPLAYPASSWORD = "";

const string DEFAULTDIR = "";

ConfigMisc::ConfigMisc()
{
    setToDefaultConfiguration();
}

string ConfigMisc::getInputGrabConfigString()
{
    if (input_grab.compare(DEFAULTINPUTGRAB)==0){return "";}
    else {return "input_grab = " + input_grab;}
}

string ConfigMisc::getInputGrabString()
{
    return input_grab;
}

string ConfigMisc::getAutomaticInputGrabConfigString()
{
    if (automatic_input_grab.compare(DEFAULTAUTOMATICINPUTGRAB)==0){return "";}
    else {return "automatic_input_grab = " + automatic_input_grab;}
}

string ConfigMisc::getAutomaticInputGrabString()
{
    return automatic_input_grab;
}

string ConfigMisc::getBsdSocketLibraryConfigString()
{
    if (bsdsocket_library.compare(DEFAULTBSDSOCKET)==0){return "";}
    else {return "bsdsocket_library = " + bsdsocket_library;}
}

string ConfigMisc::getBsdSocketLibraryString()
{
    return bsdsocket_library;
}

string ConfigMisc::getAudioBufferConfigString()
{
    if (audio_buffer_target_bytes.compare(DEFAULTAUDIOBUFFER)==0){return "";}
    else {return "audio_buffer_target_bytes = " + audio_buffer_target_bytes;}
}

string ConfigMisc::getAudioBufferString()
{
    return audio_buffer_target_bytes;
}

string ConfigMisc::getMouseSpeedConfigString()
{
    if (mouse_speed.compare(DEFAULTMOUSESPEED)==0){return "";}
    else {return "uae_input.mouse_speed = " + mouse_speed;}
}

string ConfigMisc::getMouseSpeedString()
{
    return mouse_speed;
}

string ConfigMisc::getTitleConfigString()
{
    if (title.compare(DEFAULTTITLE)==0){return "";}
    else {return "title = " + title;}
}

string ConfigMisc::getTitleString()
{
    return title;
}

string ConfigMisc::getSubTitleConfigString()
{
    if (sub_title.compare(DEFAULTSUBTITLE)==0){return "";}
    else {return "sub_title = " + sub_title;}
}

string ConfigMisc::getSubTitleString()
{
    return sub_title;
}

string ConfigMisc::getNetPlayServerConfigString()
{
    if (netplay_server.compare(DEFAULTNETPLAYSERVER)==0){return "";}
    else {return "netplay_server = " + netplay_server;}
}

string ConfigMisc::getNetPlayServerString()
{
    return netplay_server;
}

string ConfigMisc::getNetPlayPortConfigString()
{
    if (netplay_port.compare(DEFAULTNETPLAYPORT)==0){return "";}
    else {return "netplay_port = " + netplay_port;}
}

string ConfigMisc::getNetPlayPortString()
{
    return netplay_port;
}

string ConfigMisc::getNetPlayTagConfigString()
{
    if (netplay_tag.compare(DEFAULTNETPLAYTAG)==0){return "";}
    else {return "netplay_tag = " + netplay_tag;}
}

string ConfigMisc::getNetPlayTagString()
{
    return netplay_tag;
}

string ConfigMisc::getNetPlayPasswordConfigString()
{
    if (netplay_password.compare(DEFAULTNETPLAYPASSWORD)==0){return "";}
    else {return "netplay_password = " + netplay_password;}
}

string ConfigMisc::getNetPlayPasswordString()
{
    return netplay_password;
}

string ConfigMisc::getBaseDirConfigString()
{
    if (base_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "base_dir = " + base_dir;}
}

string ConfigMisc::getBaseDirString()
{
    return base_dir;
}

string ConfigMisc::getKickstartsDirConfigString()
{
    if (kickstarts_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "kickstarts_dir = " + kickstarts_dir;}
}

string ConfigMisc::getKickstartsDirString()
{
    return kickstarts_dir;
}

string ConfigMisc::getSaveStatesDirConfigString()
{
    if (save_states_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "save_states_dir = " + save_states_dir;}
}

string ConfigMisc::getSaveStatesDirString()
{
    return save_states_dir;
}

string ConfigMisc::getFloppyOverlaysDirConfigString()
{
    if (floppy_overlays_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "floppy_overlays_dir = " + floppy_overlays_dir;}
}

string ConfigMisc::getFloppyOverlaysDirString()
{
    return floppy_overlays_dir;
}

string ConfigMisc::getFlashMemoryDirConfigString()
{
    if (flash_memory_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "flash_memory_dir = " + flash_memory_dir;}
}

string ConfigMisc::getFlashMemoryDirString()
{
    return flash_memory_dir;
}

string ConfigMisc::getControllersDirConfigString()
{
    if (controllers_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "controllers_dir = " + controllers_dir;}
}

string ConfigMisc::getControllersDirString()
{
    return controllers_dir;
}

string ConfigMisc::getLogsDirConfigString()
{
    if (logs_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "logs_dir = " + logs_dir;}
}

string ConfigMisc::getLogsDirString()
{
    return logs_dir;
}

string ConfigMisc::getHardDrivesDirConfigString()
{
    if (hard_drives_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "hard_drives_dir = " + hard_drives_dir;}
}

string ConfigMisc::getHardDrivesDirString()
{
    return hard_drives_dir;
}

string ConfigMisc::getCDRomsDirConfigString()
{
    if (cdroms_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "cdroms_dir = " + cdroms_dir;}
}

string ConfigMisc::getCDRomsDirString()
{
    return cdroms_dir;
}

string ConfigMisc::getFloppiesDirConfigString()
{
    if (floppies_dir.compare(DEFAULTDIR)==0){return "";}
    else {return "floppies_dir = " + floppies_dir;}
}

string ConfigMisc::getFloppiesDirString()
{
    return floppies_dir;
}

////////////////////////////////////////////////////////////////////////////////////////////
/** \brief Checks if s is a number
      * \param s is the string to be checked
      * \return true if s is a number
      */
bool static isNumber(string s){
    istringstream buffer(s);
    int intValue;
    buffer >> intValue;
    //la conversione ritorna 0 se non è un numero. Devo stare attento al caso in cui la stringa sia proprio "0"
    if (intValue==0 && s.compare("0")==0) return true;
    return intValue!=0;
}
/** \brief Convert a string to an integer
      * \param s is the string to be converted
      * \return s as integer
      */
int static strToInt(string s){
    istringstream buffer(s);
    int intValue;
    buffer >> intValue;
    return intValue;
}
////////////////////////////////////////////////////////////////////////////

int ConfigMisc::setParameter(string parameter, string value)
{
    if(parameter.compare("input_grab")==0){
        if ((value.compare("0")==0)||(value.compare("1")==0)){
            input_grab=value;}
        else{
            input_grab=DEFAULTINPUTGRAB;
            return -1;
        }
    } else if(parameter.compare("automatic_input_grab")==0){
        if ((value.compare("0")==0)||(value.compare("1")==0)){
            automatic_input_grab=value;}
        else{
            automatic_input_grab=DEFAULTAUTOMATICINPUTGRAB;
            return -1;
        }
    } else if(parameter.compare("bsdsocket_library")==0){
        if ((value.compare("0")==0)||(value.compare("1")==0)){
            bsdsocket_library=value;}
        else{
            bsdsocket_library=DEFAULTBSDSOCKET;
            return -1;
        }
    } else if(parameter.compare("audio_buffer_target_bytes")==0){
        if (isNumber(value) && strToInt(value)>0){
            audio_buffer_target_bytes=value;
        }else{
            audio_buffer_target_bytes=DEFAULTAUDIOBUFFER;
            return -1;
        }
    } else if(parameter.compare("uae_input.mouse_speed")==0){
        if (isNumber(value) && strToInt(value)>0){
            mouse_speed=value;
        }else{
            mouse_speed=DEFAULTMOUSESPEED;
            return -1;
        }
    } else if(parameter.compare("title")==0){
        title=value;
    } else if(parameter.compare("sub_title")==0){
        sub_title=value;
    } else if(parameter.compare("netplay_server")==0){
        QStringList ipOctetsList = QString::fromStdString(value).split(".");
        if(ipOctetsList.count()!=4){
            netplay_server=DEFAULTNETPLAYSERVER;
            return -1;
        }
        for(int i=0;i<ipOctetsList.count();i++){
            if ((isNumber(ipOctetsList.at(i).toStdString())==false)||
                    (isNumber(ipOctetsList.at(i).toStdString()) && ((strToInt(ipOctetsList.at(i).toStdString())<0) || strToInt(ipOctetsList.at(i).toStdString())>255))){
                netplay_server=DEFAULTNETPLAYSERVER;
                return -1;
            }
        }
        netplay_server=value;
    } else if(parameter.compare("netplay_port")==0){
        if (isNumber(value) && strToInt(value)>0){
            netplay_port=value;
        }else{
            netplay_port=DEFAULTNETPLAYPORT;
            return -1;
        }
    } else if(parameter.compare("netplay_tag")==0){
        if (value.compare("")!=0){
            netplay_tag=value;}
        else{
            netplay_tag=DEFAULTNETPLAYTAG;
        }
    } else if(parameter.compare("netplay_password")==0){
        if (value.compare("")!=0){
            netplay_password=value;}
        else{
            netplay_password=DEFAULTNETPLAYPASSWORD;
        }
    } else if(parameter.compare("base_dir")==0){
        base_dir=value;
    } else if(parameter.compare("kickstarts_dir")==0){
        kickstarts_dir=value;
    } else if(parameter.compare("save_states_dir")==0){
        save_states_dir=value;
    } else if(parameter.compare("floppy_overlays_dir")==0){
        floppy_overlays_dir=value;
    } else if(parameter.compare("flash_memory_dir")==0){
        flash_memory_dir=value;
    } else if(parameter.compare("controllers_dir")==0){
        controllers_dir=value;
    } else if(parameter.compare("logs_dir")==0){
        logs_dir=value;
    } else if(parameter.compare("hard_drives_dir")==0){
        hard_drives_dir=value;
    } else if(parameter.compare("cdroms_dir")==0){
        cdroms_dir=value;
    } else if(parameter.compare("floppies_dir")==0){
        floppies_dir=value;
    }
    return 0;
}

void ConfigMisc::setToDefaultConfiguration()
{
    input_grab=DEFAULTINPUTGRAB;
    automatic_input_grab=DEFAULTAUTOMATICINPUTGRAB;
    bsdsocket_library=DEFAULTBSDSOCKET;
    audio_buffer_target_bytes=DEFAULTAUDIOBUFFER;
    mouse_speed=DEFAULTMOUSESPEED;
    title=DEFAULTTITLE;
    sub_title=DEFAULTSUBTITLE;

    netplay_server=DEFAULTNETPLAYSERVER;
    netplay_port=DEFAULTNETPLAYPORT;
    netplay_tag=DEFAULTNETPLAYTAG;
    netplay_password=DEFAULTNETPLAYPASSWORD;

    base_dir=DEFAULTDIR;
    kickstarts_dir=DEFAULTDIR;
    save_states_dir=DEFAULTDIR;
    floppy_overlays_dir=DEFAULTDIR;
    flash_memory_dir=DEFAULTDIR;
    controllers_dir=DEFAULTDIR;
    logs_dir=DEFAULTDIR;
    hard_drives_dir=DEFAULTDIR;
    cdroms_dir=DEFAULTDIR;
    floppies_dir=DEFAULTDIR;
}

bool ConfigMisc::hasParameter(string parameterName)
{
    return (parameterName.compare("input_grab")==0)||(parameterName.compare("automatic_input_grab")==0)||(parameterName.compare("bsdsocket_library")==0)||
            (parameterName.compare("audio_buffer_target_bytes")==0)||(parameterName.compare("title")==0)||(parameterName.compare("sub_title")==0)||
            (parameterName.compare("netplay_server")==0)||(parameterName.compare("netplay_port")==0)||(parameterName.compare("netplay_tag")==0)||
            (parameterName.compare("base_dir")==0)||(parameterName.compare("kickstarts_dir")==0)||(parameterName.compare("save_states_dir")==0)||
            (parameterName.compare("floppy_overlays_dir")==0)||(parameterName.compare("flash_memory_dir")==0)||(parameterName.compare("controllers_dir")==0)||
            (parameterName.compare("logs_dir")==0)||(parameterName.compare("hard_drives_dir")==0)||(parameterName.compare("cdroms_dir")==0)||
            (parameterName.compare("floppies_dir")==0)||(parameterName.compare("netplay_password")==0)||(parameterName.compare("uae_input.mouse_speed")==0);
}
